arangod/Pregel/Conductor.cpp

```diff
@@ -428,10 +429,6 @@ VPackBuilder Conductor::finishedWorkerStep(VPackSlice const& data) {
 
     if (_state == ExecutionState::RUNNING) {
       _startGlobalStep();  // trigger next superstep
-    } else if (_state == ExecutionState::CANCELED) {
-      LOG_PREGEL("dd721", WARN)
-          << "Execution was canceled, results will be discarded.";
-      _finalizeWorkers();  // tells workers to store / discard results
```

Comment:
If we get into the Canceled state, we do this anyway because it is part of the Canceled::run() method, which is called after a state change to this state.